Libraries/Connect/Internal/Streaming/StreamResult+ClientOnlyStream.swift

```diff
@@ -0,0 +1,54 @@
+// Copyright 2022-2024 The Connect Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+extension StreamResult {
+    /// Applies some validations which are only relevant for client-only streams.
+    ///
+    /// - parameter receivedMessageCount: The number of response messages that have been received so
+    ///                                   far, including the one being validated.
+    ///
+    /// - returns: The validated stream result, which may have been transformed into an error.
+    func validatedForClientStream(receivedMessageCount: Int) -> Self {
```

Comment:
For any Kotlin-focused code reviewers 😄 Swift defaults to `internal` so any unannotated functions are not exposed from the package